Improve configuration docs (#2411)
kedro-org/kedro: docs/source/kedro_project_setup/configuration.md

If no `env` option is specified, this will default to using the `local` environment to overwrite `conf/base`.

If, for some reason, your project does not have any other environments apart from `base`, i.e. no `local` environment to default to, you must customise the configuration loader you're using to take `default_run_env="base"` in the constructor and then specify your custom config loader subclass in `src/<package_name>/settings.py` under the `CONFIG_LOADER_CLASS` key.
Below is an example of such a custom class. If you're using the `TemplatedConfigLoader` or the `OmegaConfigLoader` you need to use either of those as the class you are subclassing.

```python

> from kedro.config import ConfigLoader

Reviewer: I know it might look a bit repetitive but is it worth adding what file are we talking about here?

Initially I thought it was `settings.py`, but then I saw that the `settings.py` is the one below this code snippet, and therefore the custom class is being defined elsewhere. Maybe we can give a suggestion of where this file can live?

Author: Done!

from typing import Any, Dict

class CustomConfigLoader(ConfigLoader):
    def __init__(
        self,
        conf_source: str,
        env: str = None,
        runtime_params: Dict[str, Any] = None,
        default_run_env: str = "base",
    ):
        super().__init__(
            conf_source=conf_source,
            env=env,
            runtime_params=runtime_params,
            default_run_env=default_run_env,
        )
```

> 
> ```python
> from my_project.custom_configloader import CustomConfigLoader
> 
> CONFIG_LOADER_CLASS = CustomConfigLoader
> ```

Reviewer: Adding a bit of clarification here:

````suggestion

And then you can import your `CustomConfigLoader` from `settings.py`:

```python
# settings.py
from my_project.custom_configloader import CustomConfigLoader

CONFIG_LOADER_CLASS = CustomConfigLoader
```
````

Since both of the file names (`parameters.yml` and `parameters_globals.yml`) match the config pattern for parameters, the `OmegaConfigLoader` will load the files and resolve the placeholders correctly.

### Custom template resolvers

> `Omegaconf` provides functionality to [register custom resolvers](https://omegaconf.readthedocs.io/en/2.3_branch/usage.html#resolvers) for templated values. You can leverage this functionality within Kedro by extending the [`OmegaConfigLoader`](/kedro.config.OmegaConfigLoader) class.

Reviewer 2: ```suggestion
`Omegaconf` provides functionality to [register custom resolvers](https://omegaconf.readthedocs.io/en/2.3_branch/usage.html#resolvers) for templated values. You can use these custom resolves within Kedro by extending the [`OmegaConfigLoader`](/kedro.config.OmegaConfigLoader) class.
```

> The example below showcases how you could do this:

Reviewer 2: ```suggestion
The example below illustrates this:
```

### Load parameters

> Parameters project configuration can be loaded with the help of any of the configuration loader classes: `ConfigLoader`, `TemplatedConfigLoader`, and `OmegaConfigLoader`.

Reviewer 2: ```suggestion
Parameters project configuration can be loaded by any of the configuration loader classes: `ConfigLoader`, `TemplatedConfigLoader`, and `OmegaConfigLoader`.
```
